meeg_preprocessing/utils.py

```diff
@@ -151,15 +153,28 @@ def setup_provenance(script, results_dir, config=None, use_agg=True):
     - sets log file for sterr output
     - writes log file with runtime information
     """
+    import inspect
+    if script is None:
+        script = inspect.stack()[1][1]
+        if script[0] == '<':
+            script = '__main__'
```

Comment:
Apparently I don't understand your workflow. When would you paste code? I never do that at this stage. And why exit ipython? google ipython autoreload.